Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -237,6 +237,40 @@ class PurchasesOrchestratorTests: StoreKitConfigTestCase {
         expect(self.backend.invokedPostReceiptDataParameters?.productData).toNot(beNil())
     }
 
+    func testPurchaseSK1PackageWithNoProductIdentifierDoesNotPostReceipt() async throws {
```

Comment:
⚠️ We didn't have any test coverage for this scenario, so we didn't know that the method in `StoreKitWorkarounds` actually made the whole purchase hang.